### PR TITLE
System test improvements

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Bugfix
- Enhancement / new feature
- ~~Refactoring~~

### Description

This PR delivers some improvements to the system tests:
* It removes the `testClusterWithCustomParameters` test which wasn't doing anything apart from checking the ConfigMap it created
* Makes `testCustomAndUpdatedValues` work on Kubernetes, not only on openShift
* `testCustomAndUpdatedValues` should cover what was probably the original idea of `testClusterWithCustomParameters`
* Removes `testDeployKafkaOnPersistentStorage` since it doesn't seem needed anymore (all other tests use persistent storage).
* Updates `replaceCm` to be able to handle multiple changes at once. Changing values one by one was causing multiple rolling updates and that costs us precious time.

### CODE REVIEW

**Comment by tombentley** [comment]
> It seems all tests were actually using persistent storage. Now all tests are using ephemeral storage and only the persistent storage tests uses persistent storage.

We started off using ephemeral storage, but it has caused us to miss certain problems in the past, so we switched to testing with persistent storage. Reverting that will reduce test coverage of the configuration we actually care about, in favour of a configuration we advertise as being only for testing.

**Comment by scholzj** [comment]
@tombentley If all tests are supposed to run on persistent ... why do we have the `testDeployKafkaOnPersistentStorage` test? My expectation and what I remebered from the past was that this test was added because all the others were supposed to be on ephemeral?

**Comment by scholzj** [comment]
@tombentley I mean if all other tests should run on persistent, I'm fine with it. But then we do not need yet another explicit test for it and can remove `testDeployKafkaOnPersistentStorage`, or?

**Comment by tombentley** [comment]
I think that test pre-dates the change to run all the tests with persistent storage.

**Comment by scholzj** [comment]
@tombentley Ok, so lets remove it then.

File: systemtest/src/test/java/io/strimzi/systemtest/AbstractClusterIT.java [hunk 1]
```diff
@@ -84,11 +86,21 @@ static String topicControllerDeploymentName(String clusterName) {
     }
 
     void replaceCm(String cmName, String fieldName, String fieldValue) {
+        Map<String, String> changes = new HashMap<>();
```
Comment (tombentley) [inline systemtest/src/test/java/io/strimzi/systemtest/AbstractClusterIT.java]: `Collections.singletonMap`

File: systemtest/src/test/java/io/strimzi/systemtest/KafkaClusterIT.java [hunk 2]
```diff
@@ -283,36 +253,38 @@ public void testForUpdateValuesInConfigMap() {
...
-        assertThat(configMap, valueOfCmEquals("zookeeper-healthcheck-timeout", "24"));
-        assertThat(configMap, valueOfCmEquals("kafka-healthcheck-delay", "23"));
-        assertThat(configMap, valueOfCmEquals("kafka-healthcheck-timeout", "20"));
-        assertThat(configMap, valueOfCmEquals("kafka-config", "{\"default.replication.factor\": 2,\"offsets.topic.replication.factor\": 2,\"transaction.state.log.replication.factor\": 2}"));
-
-        LOGGER.info("Verified CM and Testing kafka pods");
+
+        LOGGER.info("Verify values after update");
+        String configMapAfter = kubeClient.get("cm", clusterName);
+        assertThat(configMapAfter, valueOfCmEquals("zookeeper-healthcheck-delay", "31"));
```
Comment (tombentley) [inline systemtest/src/test/java/io/strimzi/systemtest/KafkaClusterIT.java]: Why did you change the values?

Comment (scholzj) [reply systemtest/src/test/java/io/strimzi/systemtest/KafkaClusterIT.java]: Honestly? I think the values should have some meaning where possible. If we think that timeout in the first place should be 10, why do we change it to something completely random such as `23`? And on top of it, if the timeout is actually used, it will take for the tests longer to succeed or fail.

Comment (tombentley) [reply systemtest/src/test/java/io/strimzi/systemtest/KafkaClusterIT.java]: OIC, you just took the existing defaults and added one to them. Makes more sense now.

Comment (scholzj) [reply systemtest/src/test/java/io/strimzi/systemtest/KafkaClusterIT.java]: Yeah, exactly, that was the idea.
